client/executor/wasmtime/src/tests.rs

```diff
@@ -109,8 +111,11 @@ impl RuntimeBuilder {
 		};
 
 		let rt = if self.precompile_runtime {
+			let dir = tempfile::tempdir().unwrap();
+			let path = dir.path().join("runtime.bin");
 			let artifact = crate::prepare_runtime_artifact(blob, &config.semantics).unwrap();
-			unsafe { crate::create_runtime_from_artifact::<HostFunctions>(&artifact, config) }
+			std::fs::write(&path, artifact).unwrap();
+			unsafe { crate::create_runtime_from_artifact::<HostFunctions>(&path, config) }
 		} else {
```

Comment:
the same concern about the interaction between `tempfile` and `create_runtime_from_artifact`